src/cascadia/TerminalApp/App.cpp

```diff
@@ -851,20 +851,38 @@ namespace winrt::TerminalApp::implementation
         {
             _lastTabClosedHandlers();
         }
+
         uint32_t tabIndexFromControl = 0;
         _tabView.Items().IndexOf(tabViewItem, tabIndexFromControl);
 
-        if (tabIndexFromControl == _GetFocusedTabIndex())
+        int focusedTabIndex = _GetFocusedTabIndex();
+
+        if (tabIndexFromControl == focusedTabIndex)
         {
-            _tabView.SelectedIndex((tabIndexFromControl > 0) ? tabIndexFromControl - 1 : 1);
+            _tabView.SelectedIndex((tabIndexFromControl > 0) ? (tabIndexFromControl - 1) : 1);
         }
 
         // Removing the tab from the collection will destroy its control and disconnect its connection.
         _tabs.erase(_tabs.begin() + tabIndexFromControl);
         _tabView.Items().RemoveAt(tabIndexFromControl);
 
         // ensure tabs and focus is sync
-        _tabView.SelectedIndex(tabIndexFromControl > 0 ? tabIndexFromControl - 1 : 0);
+        // focused tab shifts to the left if possible

```

Comment:
I'd prefer more verbose comments on each wing of this if/elseif/else. 
It requires a lot of mental work to deduce what each branch of this is doing for someone who hasn't been rooting around in the tab indexes.

Reply:
Will update